### PR TITLE
hubble: add the cluster name to a flow's source and destination endpoints

### PR DESCRIPTION
**Pull request by rolinh**
The cluster name is present in the set of labels associated to a Cilium endpoint. Adding the cluster name to a flow's source and destination endpoints will allow API clients such as Hubble UI to show if a flow is internal cluster traffic or going/coming from a different cluster (for instance, in a cluster mesh context).

```release-note
hubble: add the cluster name to a flow's source and destination endpoints
```

### CODE REVIEW

File: pkg/hubble/parser/common/endpoint.go [hunk 1]
```diff
@@ -167,19 +168,22 @@ func (r *EndpointResolver) ResolveEndpoint(ip netip.Addr, datapathSecurityIdenti
...
 	}
 	var labels []string
+	var clusterName string
 	if r.identityGetter != nil {
 		if id, err := r.identityGetter.GetIdentity(numericIdentity); err != nil {
 			r.log.WithError(err).WithField("identity", numericIdentity).
 				Debug("failed to resolve identity")
 		} else {
 			labels = SortAndFilterLabels(r.log, id.Labels.GetModel(), identity.NumericIdentity(numericIdentity))
+			clusterName = (id.Labels[k8sConst.PolicyLabelCluster]).Value
```
Comment (sayboras) [inline pkg/hubble/parser/common/endpoint.go:178]: we might not have this label for reserved identity (e.g. kube-apiserver, hosts, ingress, health, etc),  hence might need to add an existence check here.

**Review by kaworu — COMMENTED** [review]
Patch LGTM besides Tam's point about the label not being set for all endpoints.

File: pkg/hubble/parser/common/endpoint.go (hunk 1 above)
Comment (rolinh) [reply pkg/hubble/parser/common/endpoint.go:178]: `Labels` is [a map of `Label`](https://github.com/cilium/cilium/blob/962e78dfbd41ae80269ccff955f401f6f28d66a5/pkg/labels/labels.go#L166-L167), not `Label` pointers and the default value is then an empty string. The unit tests would panic if this wasn't the case.

**Review by sayboras — APPROVED** [review]

File: pkg/hubble/parser/common/endpoint.go (hunk 1 above)
Comment (kaworu) [reply pkg/hubble/parser/common/endpoint.go:178]: Ah nice, maybe worth a comment but not blocking.

**Review by kaworu — APPROVED** [review]

**Comment by kaworu** [comment]
Added the blocked label to wait on https://github.com/cilium/cilium/pull/32130 and rebase on top.

**Comment by rolinh** [comment]
> Added the blocked label to wait on #32130 and rebase on top.

#32130 is now merged, removing the blocking label.

**Pull request merged**